iothub/service/src/Amqp/AmqpConnectionHandler.cs

```diff
@@ -66,7 +72,7 @@ internal class AmqpConnectionHandler : IDisposable
         /// Returns false otherwise.
         /// </summary>
         /// <returns>True if this connection, its sessions and its sessions' links are all open. False otherwise.</returns>
-        internal bool IsOpen => _connection != null
+        internal virtual bool IsOpen => _connection != null
             && _connection.State == AmqpObjectState.Opened
```

Comment:
Made the method virtual to override the implementation for unit testing.

Reply:
I'd recommend adding a comment explaining this, otherwise it might be confusing as to why a class that isn't being inherited from in the sdk has virtual methods.

Reply:
Sure!